stl/inc/optional

```diff
@@ -445,6 +469,170 @@ public:
         return static_cast<remove_cv_t<_Ty>>(_STD forward<_Ty2>(_Right));
     }
 
+#if _HAS_CXX23
+    template <class _Fn>
+    constexpr auto and_then(_Fn&& _Func) & {
+        using _Uty = invoke_result_t<_Fn, _Ty&>;
+
+        static_assert(_Is_specialization_v<remove_cvref_t<_Uty>, optional>,
+            "optional<T>::and_then(F) requires the return type of F to be a specialization of optional "

```

Comment:
With the `remove_cvref_t`, this is more "requires the type of `f(value())` (where `f` denotes the argument to `and_then`) to be a cv-qualified specialization of `optional`" but that's too precise to be clear to the people who actually need this message. Maybe "requires the argument to return a specialization of `optional`"?

No change required if you don't feel it's an improvement, but if you do please change all four places consistently.